drake/systems/plants/RigidBodySystem.h

```diff
@@ -0,0 +1,125 @@
+#ifndef DRAKE_RIGIDBODYSYSTEM_H
+#define DRAKE_RIGIDBODYSYSTEM_H
+
+#include "RigidBodyTree.h"
+#include "RigidBodyIK.h"  // required for resolving initial conditions (could be moved to a cpp file)
+
+namespace Drake {
+
+  class RigidBodySystem {
+  public:
+    template <typename ScalarType> using InputVector = Eigen::Matrix<ScalarType,Eigen::Dynamic,1>;
+    template <typename ScalarType> using StateVector = Eigen::Matrix<ScalarType,Eigen::Dynamic,1>;
+    template <typename ScalarType> using OutputVector = Eigen::Matrix<ScalarType,Eigen::Dynamic,1>;
```

Comment:
I still really like the idea of having the KinematicsCache be the state vector for RigidBodySystem.